lettuce/streaming.py

```diff
@@ -7,15 +7,38 @@
 
 
 class StandardStreaming:
-    """Standard Streaming step on a regular grid."""
+    """Standard Streaming step on a regular grid.
+
+    Attributes
+    ----------
+    no_stream_mask : torch.Tensor
+        Boolean mask with the same shape as the distribution function f.
+        If None, stream all (also around all boundaries).
+    """
     def __init__(self, lattice):
         self.lattice = lattice
+        self._no_stream_mask = None
+
+    @property
+    def no_stream_mask(self):
+        return self._no_stream_mask
+
+    @no_stream_mask.setter
+    def no_stream_mask(self, mask):
+        self._no_stream_mask = mask
 
     def __call__(self, f):
-        for i in range(self.lattice.Q):
-            f[i] = torch.roll(f[i], shifts=tuple(self.lattice.stencil.e[i]), dims=tuple(np.arange(self.lattice.D)))
+        for i in range(1, self.lattice.Q):
+            if self.no_stream_mask is None:
+                f[i] = self._stream(f, i)
+            else:
+                new_fi = self._stream(f, i)
```

Comment:
Doesn't this use twice the memory? (for the one i?)
I had tried putting the roll directly into the where, though I'm not sure if that is any better:

`f[i] = torch.where(self.no_streaming_mask[i], f[i], torch.roll(f[i], shifts=tuple(self.lattice.stencil.e[i]), dims=tuple(np.arange(self.lattice.D))))`

Reply:
not sure. didn't test. I think this behaves like the previous simpler code but you can try if it makes a difference.

Anyway, you need to make some kind of copy. Otherwise you cannot distinguish between the rolled fi and the original.